### DISCUSSION TITLE
SB and Geolocation

### DISCUSSION THREAD

**Discussion by Allmight3** [seleniumbase/SeleniumBase#2903] [Q&A] [answered]
I'm unable to get my geolocation to work, as it seems to be disabled/blocked by default.

```
from seleniumbase import SB

def browser_launch():
	brave_location = "/Applications/Brave Browser.app/Contents/MacOS/Brave Browser"
	with SB(uc=True, binary_location=brave_location, incognito=True, user_data_dir="SBProfile") as sb:
		sb.open("about:blank")
		sb.execute_cdp_cmd(
			"Browser.grantPermissions",
			{
				"origin": "https://webbrowsertools.com/geolocation/",
				"permissions": ["geolocation"],
			},
		)
		sb.execute_cdp_cmd(
			"Emulation.setGeolocationOverride",
			{
				"latitude": 40.730610,
				"longitude": -73.935242,
				"accuracy": 100,
			},
		)
		sb.driver.get("https://webbrowsertools.com/geolocation/")
		sb.assert_text("N/A", selector="#altitude", by="css selector", timeout=15)

browser_launch()
```

<img width="547" alt="Screenshot 2024-07-04 at 3 55 06 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/35853584/208ac782-11f9-497a-be73-6adb7bbbe411">

I tried various things, including not using incognito. I also tried not using Brave, and instead just using Chrome. However those all resulted in the same thing. I also tried `sb.driver.execute_cdp_cmd()` however that faired no better.

The geolocation example from the docs ( examples/test_geolocation.py ) does work, even when I change the website to this one. Am I mis-using SB, or is this not possible with SB?

**Comment by mdmintz** [comment, marked as answer]
The GeoLocation example: [SeleniumBase/examples/test_geolocation.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_geolocation.py).

When the driver is disconnected (which is needed to make chromedriver undetectable) any `execute_cdp_cmd` changes are reset (because those changes are effective only while the driver is connected).

To make the GeoLocation changes persistent, you need to use a Chrome extension. But note that you can't add extensions when you browse in Incognito mode or as a guest. (https://support.google.com/chrome_webstore/answer/2664769?hl=en )
